activesupport/lib/active_support/xml_mini/rexml.rb

```diff
@@ -20,11 +20,9 @@ def parse(data)
         data = StringIO.new(data || '')
       end
 
-      char = data.getc
-      if char.nil?
+      if data.eof?
```

Comment:
It means there is no more data to read, detalis here: [IO#eof?](http://ruby-doc.org/core-2.3.0/IO.html#method-i-eof-3F). 
Returns true if getc returns nil [getc](http://ruby-doc.org/core-2.3.0/IO.html#method-i-getc), but will not eat a char.